### PR TITLE
Abort signing when signing timeout has passed

### PR DESCRIPTION
**Pull request by nkuba**
The client was not checking if signing has timed out so it was trying to publish the signature over and over. Here we update the client to check if the signing timeout has been hit in the contract so there is no point in publishing the signature or starting signature calculation on client start.

Closes: https://github.com/keep-network/keep-ecdsa/issues/404

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -295,7 +295,23 @@ func (n *Node) publishSignature(
...
+		hasSigningTimedOut, err := n.ethereumChain.HasSigningTimedOut(keepAddress)
+		if err != nil {
+			logger.Errorf(
+				"failed to verify if signing has timed out for keep [%s]: [%v]",
+				keepAddress.String(),
+				err,
+			)
+			continue
+		}
+		if !hasSigningTimedOut {
```
Comment (pdyraga) [inline pkg/node/node.go:313]:
```
if hasSigningTimedOut
```
should it be.

File: pkg/chain/chain.go [hunk 2]
```diff
@@ -113,6 +114,9 @@ type BondedECDSAKeep interface {
 	// calculated for the given digest.
 	IsAwaitingSignature(keepAddress common.Address, digest [32]byte) (bool, error)
 
+	// HasSigningTimedOut checks if signing has timed out for keep.
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply pkg/chain/chain.go]: Done in https://github.com/keep-network/keep-ecdsa/pull/412/commits/2c65e240dd15af752d1df2a85011886c96306e49

File: pkg/node/node.go (hunk 1 above)
Comment (pdyraga) [reply pkg/node/node.go]: Done in https://github.com/keep-network/keep-ecdsa/pull/412/commits/755c8ba8e6432c0f458153df223cd12643b197c5

**Comment by pdyraga** [comment]
Looks good to me now but I did two changes so I'll leave the final decision about merging to you @nkuba.

**Comment by nkuba** [comment]
Oh yeah, thanks for the updates dude!

**Comment by nkuba** [comment]
Let's ship it!

**Comment by pdyraga** [comment]
Shippin'

**Pull request merged**